builtin/logical/kv/path_metadata.go

```diff
@@ -486,14 +511,33 @@ func (b *versionedKVBackend) pathMetadataWrite() framework.OperationFunc {
 		if err != nil {
 			return nil, err
 		}
+
+		now := ptypes.TimestampNow()
 		if meta == nil {
-			now := ptypes.TimestampNow()
 			meta = &KeyMetadata{
```

Comment:
I guess it doesn't truly matter since we have granular locking, but it seems like it'd be best to have CAS checked here as well and prevent the request if the caller sets `Metadata >= 1`  since that implies they think it already exists. Otherwise, you could have an inbound request with `metadata == 53` and a delete call which just beats it, causing the write to create a new entry.